frontend/app/src/entities/events/domain/get-events.query.ts

```diff
@@ -1,9 +1,21 @@
 import { OBJECTS_PER_PAGE } from "@/entities/events/api/get-events-from-api";
 import { GetEventsParams, getEvents } from "@/entities/events/domain/get-events";
-import { infiniteQueryOptions, useInfiniteQuery } from "@tanstack/react-query";
+import {
+  UseInfiniteQueryOptions,
+  infiniteQueryOptions,
+  useInfiniteQuery,
+} from "@tanstack/react-query";
 
-export function getEventsQueryOptions({ filters }: GetEventsParams) {
+interface GetEventsQueryOptions extends GetEventsParams {
+  queryOptions: Omit<
+    UseInfiniteQueryOptions,
+    "queryKey" | "queryFn" | "initialPageParam" | "getNextPageParam"
+  >;
```

Comment:
there is an utility types for it: `QueryConfig` you can see an example of usage in useGetTaskCount